src/osd/PrimaryLogPG.cc

```diff
@@ -5036,11 +5036,11 @@ struct FillInVerifyExtent : public Context {
     r(r), rval(rv), outdatap(blp), maybe_crc(mc),
     size(size), osd(osd), soid(soid), flags(flags) {}
   void finish(int len) override {
-    *r = len;
     if (len < 0) {
       *rval = len;
       return;
     }
+    *r = len;
```

Comment:
how about also setting `*r` to zero in the above `if` block?

Reply:
hi @tchaikov, Sorry for late reply.
To consider that op would re-queue, maybe we should keep extent length in op even if get error?

Reply:
hi @viCEnTECheNg , indeed. thanks for pointing this out.